crates/holochain/src/core/workflow/sys_validation_workflow.rs

```diff
@@ -43,170 +42,424 @@ mod validate_op_tests;
 
 #[instrument(skip(
     workspace,
-    space,
+    current_validation_dependencies,
     trigger_app_validation,
-    sys_validation_trigger,
+    trigger_self,
     network
 ))]
 pub async fn sys_validation_workflow<Network: HolochainP2pDnaT + Clone + 'static>(
     workspace: Arc<SysValidationWorkspace>,
-    space: Arc<Space>,
+    current_validation_dependencies: Arc<Mutex<ValidationDependencies>>,
     trigger_app_validation: TriggerSender,
-    sys_validation_trigger: TriggerSender,
+    trigger_self: TriggerSender,
     network: Network,
 ) -> WorkflowResult<WorkComplete> {
-    let complete =
-        sys_validation_workflow_inner(workspace, space, network, sys_validation_trigger).await?;
+    // Run the actual sys validation using data we have locally
+    let outcome_summary =
+        sys_validation_workflow_inner(workspace.clone(), current_validation_dependencies.clone())
+            .await?;
+
+    // trigger app validation to process any ops that have been processed so far
+    if outcome_summary.accepted > 0 {
+        tracing::debug!("Sys validation accepted {} ops", outcome_summary.accepted);
+
+        trigger_app_validation.trigger(&"sys_validation_workflow");
+    }
 
-    // --- END OF WORKFLOW, BEGIN FINISHER BOILERPLATE ---
+    // Now go to the network to try to fetch missing dependencies
```

Comment:
What is the plan for this part, will we see how it plays out in practice to fetch dependencies as part of the sys validation and refactor in case it's slow? Or do you have the same in mind for a follow up refactor as for app validation, where dependencies gets are fired and forgotten?

Reply:
Yeah, I think this needs more work. There's no guarantee that we get the action we need via publishing in a reasonable time frame. We do genuinely want to retry this. But it would be cleaner to pull this out to somewhere else and not block the workflow with it. A background Tokio task would probably do the trick. 

That will somewhat change things around how this workflow gets re-triggered... so possible this will need another look before we start back porting anything to avoid making a mess going back and forth on 0.2...

Reply:
> "Never wait for network requests. Instead, register a callback to be run when the requested data is available."

Yes, that's what I'm thinking too. The cache could serve as a source of validation triggers. For both sys and app validation we can monitor the cache and which deps we're expecting, and once they arrive the corresponding workflow can be triggered.